bwreg-webapp/src/main/webapp/service/set-password.xhtml

```diff
@@ -52,18 +52,20 @@
 	</p:panel>
 
 	<h:panelGroup rendered="#{not setServicePasswordBean.navOptions.equalsIgnoreCase('created')}">
-		<h3><h:outputText value="#{messages.delete_password} #{setServicePasswordBean.serviceEntity.name}" /></h3>
-	
-		<p:panel>		
-			<div style="margin-bottom: 1em;">
-				<h:outputText value="#{messages.service_password_help_delete}" escape="false" />
-			</div>
-			<div>
-				<p:commandButton id="delete" action="#{setServicePasswordBean.deleteServicePassword()}" value="#{messages.delete}"
-					update="messageBox,baseData" />
-			</div>
-		</p:panel>
+		<h:panelGroup rendered="#{not setServicePasswordBean.hideDeletePasswordBtn}">

```

Comment:
I change the <div> with the display css for a <h:panelGroup>. It just lets JSF do the dirty work, for example, the update property in the button. If JSF doesn't render the panelGroup it automatically deletes all events inside.